### PR TITLE
fixed missing dep and added testing guide

### PR DESCRIPTION
**Pull request by cmishra**
- added missing dep 
 - added testing guide (very high level, largely describes the current process) 
 - added dependency list for notebooks (to enable automation down the line)

### CODE REVIEW

**Review by gjoliver — APPROVED** [review]
nice

File: pyproject.toml [hunk 1]
```diff
@@ -35,6 +35,7 @@ dependencies = [
   "attrs",
   "pandas",
   "cloudpathlib",
+  "tenacity"
```
Comment (ebetica) [inline pyproject.toml:38]: Let's keep # of dependencies low? I think we can pretty much rewrite this.

Comment (cmishra) [reply pyproject.toml:38]: Based on internal discussions decided to keep

File: CONTRIBUTIONS.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
+We welcome community contributions to help make this package better!
+
+## Contributing
+
+_to be written_
```
[the remark this answers is not in the document]
Comment (cmishra) [reply CONTRIBUTIONS.md:5]: Maybe, but you or Raul would be better candidates to do so than I :-). Will merge this as-is for now.

**Pull request merged**